describe/LN.spec.ts

```diff
@@ -0,0 +1,183 @@
+        <DOType id="WYE2" cdc="WYE">
+          <SDO name="phsA" type="CMV2" />
+        </DOType>
+        <DOType cdc="CMV" id="CMV">
+          <DA name="cVal" bType="Struct" fc="MX" type="Vector"/>
+        </DOType>
+        <DOType cdc="CMV" id="CMV2">
+          <DA name="cVal" bType="Struct" fc="MX" type="Vector2"/>
+        </DOType>
+        <DAType id="Vector" >
+          <BDA name="mag" bType="Struct" type="AnalogueValue" />
+        </DAType>
+        <DAType id="Vector2" >
+          <BDA name="mag" bType="Struct" type="AnalogueValue2" />
+        </DAType>
+        <DAType id="AnalogueValue" >
+          <BDA name="f" bType="FLOAT32" >
+              <Val sGroup="3">60.60</Val>
+              <Val sGroup="1">10.10</Val>
+              <Val sGroup="2">40.10</Val>
+          </BDA>
+        </DAType>
+        <DAType id="AnalogueValue2" >
+          <BDA name="f" bType="FLOAT32" >
+              <Val sGroup="3">30.30</Val>
+              <Val sGroup="1">10.10</Val>
+              <Val sGroup="2">20.20</Val>
+          </BDA>
+        </DAType>
+        <EnumType id="BehModKind" >
+          <EnumVal ord="1">on</EnumVal>
+          <EnumVal ord="3">test</EnumVal>
+          <EnumVal ord="5">off</EnumVal>
+        </EnumType>
+      </DataTypeTemplates>
+    </SCL>`,
+  "application/xml"
+);
+
+const missingLnType = scl.querySelector('*[desc="missingLnType"]')!;
+const invalidLnType = scl.querySelector('*[desc="invalidLnType"]')!;
+const invalidLnTypeDescription = scl.querySelector(
+  '*[desc="invalidLnTypeDescription"]'
+)!;
+
+const baseLLN0 = scl.querySelector(
+  `LDevice[inst="lDevice1"] > *[lnClass="LLN0"][inst=""]`
+)!;
+const equalLLN0 = scl.querySelector(
+  'LDevice[inst="lDevice2"] > *[lnClass="LLN0"][inst=""]'
+)!;
+const diffLLN0 = scl.querySelector(
+  'LDevice[inst="lDevice3"] > *[lnClass="LLN0"][inst=""]'
+)!;
+const baseMMXU = scl.querySelector(
+  `LDevice[inst="lDevice1"] > *[lnClass="MMXU"][inst="1"]`
+)!;
+const equalMMXU = scl.querySelector(
+  `LDevice[inst="lDevice2"] > *[lnClass="MMXU"][inst="1"]`
+)!;
+const diffMMXU = scl.querySelector(
+  `LDevice[inst="lDevice3"] > *[lnClass="MMXU"][inst="1"]`
+)!;
+
+describe("Description for SCL schema type LN", () => {
+  it("returns undefined with missing lnType attribute", () =>
+    expect(LN(missingLnType)).to.be.undefined);
+
+  it("returns undefined with invalid lnType attribute", () =>
+    expect(LN(invalidLnType)).to.be.undefined);
+
+  it("returns undefined with invalid LNodeType description", () =>
+    expect(LN(invalidLnTypeDescription)).to.be.undefined);
+
+  it("return logical node structure in lnType key", () =>
+    expect(LN(baseLLN0)?.lnType).to.satisfy(isLNodeTypeDescription));
+
+  it("returns same description with semantically equal LN's", () => {
+    expect(JSON.stringify(LN(baseLLN0))).to.equal(
+      JSON.stringify(LN(equalLLN0))
+    );
+    expect(JSON.stringify(LN(baseMMXU))).to.equal(
+      JSON.stringify(LN(equalMMXU))
+    );
+  });
```

Comment:
Can you think of more relevant cases to be tested than just these two here? I'm curious whether an `LNodeType` might look different on the surface from another but really represent the same data structure in more ways than just this. Do we have all the other cases covered by tests for elements lower down in the hierarchy (DOType, DAType and EnumType)? My question is a bit murky here, maybe I don't even have a real question, it's just that intuitively testing only these two cases seemed a bit sparse to me. Maybe it's best we talk over the possible cases for LN meaning differences again, just to be sure nothing terribly important was missed here.

Reply:
I have added some diff tests for different SDO/DO/DA/BDA and EnumType element within a `lNodeTypeDescription`.